Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -1,50 +1,84 @@
+- **Flavor:** Debug, Release
+- **Architecture:** x86, x64, ARM, ARM64, AnyCPU
+- **OS:** Windows_NT, Linux, OSX, FreeBSD, AnyOS
+- **Target Frameworks:** NetFx (aka Desktop), netstandard (aka dotnet/Portable), NETCoreApp (aka .NET Core), UAP (aka UWP/Store/netcore50), [??Should we have any target framework??]
+- **Platform Runtimes:** [??Should we remove this option??] NetFx (aka CLR/Desktop), CoreCLR, CoreRT (aka NetNative/AOT/MRT)
+
+##Individual build properties
+The following are the properties associated with each build pivot
+
+- `$(ConfigurationGroup) -> empty or Debug* or Release`
+- `$(Platform) -> empty/AnyCPU* or x86 or x64 or ARM or ARM64`
+- `$(OSGroup) -> empty/AnyOS* or Windows or Linux or OSX or FreeBSD`
+- `$(TargetGroup) -> $(PackageTargetFramework)$(PackageTargetRuntime)`
+ - `$(PackageTargetFramework) -> netstandard1.x or netcoreapp1.x or net46x or uap10.x`
+ - `$(PackageTargetRuntime) -> empty or aot`
+ - For more information on various targets see also [.NET Standard](https://github.com/dotnet/standard/blob/master/docs/versions.md)
 
-In the pkg directory for the library there should be only **one** `.pkgproj` for the primary package for the library.  If the library has platform-specific implementations those should be split into platform specific projects in a subfolder for each platform. (see [Package projects](./package-projects.md))
+<BR/>`*` -> *default values*
 
-##Build Pivots
-Below is a list of all the various options we pivot the project builds on.
+##Aggregate build properties
+Each project will define a set of supported build configurations
 
-- **Architecture:** x86, x64, ARM, ARM64
-- **Flavor:** Debug, Release
-- **OS:** Windows_NT, Linux, OSX, FreeBSD
-- **Platform Runtimes:** NetFx (aka CLR/Desktop), CoreCLR, CoreRT (aka NetNative/AOT/MRT)
-- **Target Frameworks:** NetFx (aka Desktop), netstandard (aka dotnet/Portable), NETCoreApp (aka .NET Core), UAP (aka UWP/Store/netcore50)
-- **Version:** Potentially multiple versions at the same time.
-- **TestTFM:** net46, netcoreapp1.1, netcore50. This is the TFM that will be used to run the tests in. (Used in test projects only)
+```
+<PropertyGroup>
+  <BuildConfigurations>
+    [BuildConfiguration]
+  </BuildConfigurations>
+<PropertyGroup>
+```
 
-##Full Repo build pass
-**Build Parameters:** *Flavor, Architecture*<BR/>
-For each combination of build parameters there should be a full build pass over the entire repo.
+- `BuildConfiguration -> $(Configuration)|$(Platform)`
+- `$(Configuration) -> $(OSGroup)-$(TargetGroup)-$(ConfigurationGroup)`
```

Comment:
Okay, as long as this still helps VS understand the configurations, then it's okay. I wasn't sure it would, though, since it's formatted slightly different than the ones we have right now (conditions on a PropertyGroup).